SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/AlwaysInterestedTest/AlwaysInterestedTest.cpp

```diff
@@ -0,0 +1,335 @@
+			},
+			StepTimeLimit);
+	}
+
+	{ // Step 10 - Validate visibility on server 1
+		AddStep(
+			TEXT("AlwaysInterested_ValidateVisibilityOnServer1_Part2"), FWorkerDefinition::Server(1), nullptr, nullptr,
+			[this](float DeltaTime) {
+				RequireTrue(IsValid(ActorWithAlwaysInterestedProperty), TEXT("Should see root actor via authority"));
+				RequireTrue(!IsValid(InterestedInThisReplicatedActor), TEXT("Shouldn't see interested actor"));
+				RequireTrue(!IsValid(NotInterestedInThisReplicatedActor), TEXT("Shouldn't see not-interested actor"));
+				RequireTrue(IsValid(OtherInterestedInThisReplicatedActor), TEXT("Should see other interested actor via interest"));
+				FinishStep(); // This will only actually finish if requires are satisfied
+			},
+			StepTimeLimit);
+	}
+
+	{ // Step 11 - Move actors back to original server
+		AddStep(TEXT("AlwaysInterestedMoveActorsBack"), FWorkerDefinition::Server(2), nullptr, [this]() {
+			// Move both interested actors back to server 1's area
+			AssertTrue(InterestedInThisReplicatedActor->HasAuthority(), TEXT("Must have authority over InterestedInThisReplicatedActor"));
+			AssertTrue(NotInterestedInThisReplicatedActor->HasAuthority(),
+					   TEXT("Must have authority over NotInterestedInThisReplicatedActor"));
+			AssertTrue(OtherInterestedInThisReplicatedActor->HasAuthority(),
+					   TEXT("Must have authority over OtherInterestedInThisReplicatedActor"));
+			InterestedInThisReplicatedActor->SetActorLocation(OtherWorkerPosition);
+			NotInterestedInThisReplicatedActor->SetActorLocation(OtherWorkerPosition);
+			OtherInterestedInThisReplicatedActor->SetActorLocation(OtherWorkerPosition);
+
+			FinishStep();
+		});
+	}
+
+	{ // Step 12 - Validate visibility on server 1 again
+		AddStep(
+			TEXT("AlwaysInterested_ValidateVisibilityOnServer1_Part3"), FWorkerDefinition::Server(1), nullptr, nullptr,
+			[this](float DeltaTime) {
+				RequireTrue(IsValid(ActorWithAlwaysInterestedProperty), TEXT("Should see root actor via authority"));
+				// There's currently an issue where if an authoritative server loses interest over an actor reference, it won't
+				// correct the reference when the actor comes back into view.
+				// RequireTrue(IsValid(InterestedInThisReplicatedActor), TEXT("Should see interested actor via authority"));
+				// RequireTrue(IsValid(NotInterestedInThisReplicatedActor), TEXT("Should see not-interested actor via authority"));
+				RequireEqual_Int(GetNumberOfActorsOfType<ASmallNCDActor>(GetWorld()), 3, TEXT("Should see all actors via authority"));
+				RequireTrue(IsValid(OtherInterestedInThisReplicatedActor), TEXT("Should see other interested actor via authority"));
+				FinishStep(); // This will only actually finish if requires are satisfied
+			},
+			StepTimeLimit);
+	}
+
+	{ // Step 13 - Validate visibility on server 2 again
+		AddStep(
+			TEXT("AlwaysInterested_ValidateVisibilityOnServer2_Part3"), FWorkerDefinition::Server(2), nullptr, nullptr,
```

Comment:
It may be worth refactoring into e.g.
`RequireVisibilityStep(Server(2), false, false, false, false);`

Reply:
but I guess this has more useful error msges

Reply:
yeah think i'll leave it as in for verbosity reasons